operator/cnp_event.go

```diff
@@ -59,7 +79,21 @@ func enableCNPWatcher() error {
 							return
 						}
 
-						groups.UpdateDerivativeCNPIfNeeded(newCNP.CiliumNetworkPolicy, oldCNP.CiliumNetworkPolicy)
+						if !newCNP.RequiresDerivative() {
```

Comment:
The old `UpdateDerivativeCNPIfNeeded` function would also update the local copy via `addDerivativeCNP`, similar to the `AddFunc` callback just above. The code expanded here deletes old CNPs if they are updated and the new copy doesn't need a derivative. It never handles the case where newCNP changes something but still requires a derivative.

Reply:
Seems to me like `addDerivativeCNP` is always called in controller update if `newCNP.RequiresDerivative` is true